STM32F1/cores/maple/libmaple/timer.c

```diff
@@ -215,7 +215,8 @@ void timer_disable(timer_dev *dev) {
  * @param mode New timer mode for channel
  */
 void timer_set_mode(timer_dev *dev, uint8 channel, timer_mode mode) {
-    ASSERT_FAULT(channel > 0 && channel <= 4);

```

Comment:
This is the origin of this PR, this `ASSERT_FAULT(channel > 0 && channel <= 4); `code triggered when the old example sent in channel = 0.  
I changed it so that the assert would ignore channel when in encoder mode. But I did a rollback after you guys suggested i should correct the example instead.

ASSERT_FAULT triggers when condition is false, just like any ASSERT

STM32F1/system/libmaple/include/libmaple/util.h:
```
#define ASSERT_FAULT(exp)                       \
    if (exp) {                                  \
    } else {                                    \
        _fail(__FILE__, __LINE__, #exp);        \
    }
```